src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/TransportCreation/TransportCreationHandler.cs

```diff
@@ -3,48 +3,34 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
-using System.Net;
-using System.Net.Sockets;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.Data.SqlClient;
-using Microsoft.Data.SqlClient.SNI;
 
 namespace Microsoft.Data.SqlClientX.Handlers.TransportCreation
 {
-    internal sealed class TransportCreationHandler : IHandler<ConnectionHandlerContext>
+    /// <summary>
+    /// Handler for creating the connection to the server. This handler passes the context through
+    /// its own chain of responsibilities to determine what protocol to connect with.
+    /// </summary>
+    internal sealed class TransportCreationHandler : ContextHandler<ConnectionHandlerContext>
     {
-        private const int KeepAliveIntervalSeconds = 1;
-        private const int KeepAliveTimeSeconds = 30;
+        private readonly ReturningHandler<ConnectionHandlerContext, Stream> _streamCreationChain;
 
-        #if NET8_0_OR_GREATER
-        private static readonly TimeSpan DefaultPollTimeout = TimeSpan.FromSeconds(30);
-        #else
-        private const int DefaultPollTimeout = 30 * 100000; // 30 seconds as microseconds
-        #endif
-
-        /// <inheritdoc />
-        public IHandler<ConnectionHandlerContext> NextHandler { get; set; }
+        public TransportCreationHandler()
+        {
+            // Construct the chain of responsibility for handling the connections
+            _streamCreationChain = new SharedMemoryTransportCreationHandler();
+            _streamCreationChain.NextHandler = new TcpTransportCreationHandler();
+            _streamCreationChain.NextHandler.NextHandler = new NamedPipeTransportCreationHandler();
+        }
 
         /// <inheritdoc />
-        public async ValueTask Handle(ConnectionHandlerContext context, bool isAsync, CancellationToken ct)
+        public override async ValueTask Handle(ConnectionHandlerContext context, bool isAsync, CancellationToken ct)
         {
-            Debug.Assert(context.DataSource is not null, "context.DataSource is null");
-
             try
             {
-                // @TODO: Build CoR for handling the different protocols in order
-                if (context.DataSource.ResolvedProtocol is DataSource.Protocol.TCP)
-                {
-                    context.ConnectionStream = await HandleTcpRequest(context, isAsync, ct).ConfigureAwait(false);
-                }
-                else
-                {
-                    throw new NotImplementedException();
-                }
+                context.ConnectionStream = await _streamCreationChain.Handle(context, isAsync, ct);
```

Comment:
Do we want `.ConfigureAwait(false)` here?

Reply:
Yep, makes sense to me